proposals/nnnn-addcommand-ui-updates.md

```diff
@@ -0,0 +1,166 @@
+```xml
+<enum name="ImageFieldName" since="3.0">
+    <!-- New Parameters -->
+    <element name="menuCommandSecondaryImage" since = "x.x">
+        <description>The secondary image field for AddCommand</description>
+    </element>
+    <element name="menuSubmenuSecondaryImage" since = "x.x">
+        <description>The secondary image field for AddSubmenu</description>
+    </element>
+</enum>
+```
+
+### HMI_API Updates
+Updates also need to be made to the HMI_API so that Core can communicate with the HMI about the new types.
+
+###### AddSubmenu
+```xml
+<function name="UI.AddSubMenu" messagetype="request">
+    <!-- New Parameters -->
+    <param name="secondaryText" maxlength="500" type="String" mandatory="false">
+        <description>Optional secondary text to display</description>
+    </param>
+    <param name="tertiaryText" maxlength="500" type="String" mandatory="false">
+        <description>Optional tertiary text to display</description>
+    </param>
+    <param name="secondaryImage" type="Image" mandatory="false">
+        <description>Optional secondary image struct for menu cell</description>
+    </param>
+</function>
+```
+
+###### AddCommand
+```xml
+<function name="UI.AddCommand" messagetype="request">
+    <!-- New Parameters -->
+    <param name="secondaryText" maxlength="500" type="String" mandatory="false">
+        <description>Optional secondary text to display</description>
+    </param>
+    <param name="tertiaryText" maxlength="500" type="String" mandatory="false">
+        <description>Optional tertiary text to display</description>
+    </param>
+    <param name="secondaryImage" type="Image" mandatory="false">
+        <description>Optional secondary image struct for menu cell</description>
+    </param>
+</function>
+```
+
+### High-Level Manager Updates
+Additional changes will need to be made to the high-level managers to account for these RPC changes.
+
+#### iOS Library
+```objc
+@interface SDLMenuCell: NSObject
+// New Parameters
+@property (copy, nonatomic, readonly, nullable) NSString *secondaryText;
+@property (copy, nonatomic, readonly, nullable) NSString *tertiaryText;
+@property (strong, nonatomic, readonly, nullable) SDLArtwork *secondaryArtwork;
+
+@end
+```
+
+#### Java Library
+```java
+public class MenuCell implements Cloneable{
+    private String secondaryText, tertiaryText;
+    private SdlArtwork secondaryArtwork;
+
+    // Provide appropriate getters, setters, and initializers
+}
+```
+
+#### General Notes
+The manager layer will have to strip any data that cannot be sent (e.g. if the head unit doesn't support `tertiaryText`, then don't send that field).
+
+## Potential downsides
+The primary potential downside is that this introduces more variability between versions and HMIs. Between versions because older head units won't support these new UI elements. Between HMIs because HMIs may not place these UI elements in the same places, but this downside is shared with all SDL template UIs.
```

Comment:
Please update to this:

The primary potential downside is that this introduces more variability between versions and HMIs. Between versions because older head units won't support these new UI elements and, between HMIs because HMIs may not place these UI elements in the same places. But, this downside is shared with all SDL template UIs.

Reply:
Thanks for modifying this.  
For better readability, please make the second and third not start with Between

Thanks